packages/byods/src/types.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Represents the configuration options for the SDK.
+ * @interface SDKConfig
+ * @property {string} clientId - The client ID.
+ * @property {string} clientSecret - The client secret.
+ * @property {string} [baseUrl='https://webexapis.com/v1'] - The base URL for the API requests.
+ */
+export interface SDKConfig {
+  clientId: string;
+  clientSecret: string;
+  baseUrl?: string;
+}
+
+/**
+ * TokenResponse JSON shape from Webex APIs.
+ * @interface
+ * @property {string} access_token - The access token.
+ * @property {number} expires_in - The expiration time of the access token in seconds.
+ * @property {string} refresh_token - The refresh token.
+ * @property {number} refresh_token_expires_in - The expiration time of the refresh token in seconds.
+ * @property {string} token_type - The type of the token.
+ */
+export interface TokenResponse {
+  access_token: string;
+  expires_in: number;
+  refresh_token: string;
+  refresh_token_expires_in: number;
+  token_type: string;
+}
+
+/**
+ * Represents a token with its expiration details.
+ * @interface
+ * @property {string} accessToken - The access token.
+ * @property {string} refreshToken - The refresh token.
+ * @property {Date} expiresAt - The expiration date of the access token.
+ * @property {Date} refreshAccessTokenExpiresAt - The expiration date of the refresh token.
+ */
+export interface Token {
+  accessToken: string;
+  refreshToken: string;
+  expiresAt: Date;
+  refreshAccessTokenExpiresAt: Date;
+}
+
+/**
+ * Represents a service app authorization for an organization.
+ * @interface
+ * @property {string} orgId - The organization ID.
+ * @property {Token} token - The token details.
+ */
+export interface ServiceAppAuthorization {
+  orgId: string;
+  token: Token;
+}
+
+/**
+ * Represents a collection of service app authorizations.
+ * @interface
+ * @property {string} [orgId: string] - The organization ID mapped to its authorization details.
+ */
+export interface ServiceAppAuthorizations {
+  [orgId: string]: ServiceAppAuthorization;
+}
+
+export interface ApiResponse<T> {
+  data: T;
+  status: number;
+}
+
+/**
```

Comment:
I think this deserves a separate types file. Perhaps even a separate folder under src/httpClient

Reply:
Moved to a dedicated folder